pkg/clusters/addons/loadimage/addon.go

```diff
@@ -0,0 +1,73 @@
+package loadimage
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters/types/kind"
+)
+
+// -----------------------------------------------------------------------------
+// CertManager Addon
+// -----------------------------------------------------------------------------
+
+const (
+	// AddonName indicates the unique name of this addon.
+	AddonName clusters.AddonName = "loadimage"
+)
+
+type Addon struct {
+	image  string
+	loaded bool
+}
+
+func New() clusters.Addon {
+	return &Addon{}
+}
+
+// -----------------------------------------------------------------------------
+// CertManager Addon - Addon Implementation
+// -----------------------------------------------------------------------------
+
+func (a *Addon) Name() clusters.AddonName {
+	return AddonName
+}
+
+func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
+	if a.image == "" {
+		a.loaded = true
+		return nil
+	}
```

Comment:
Why is this here - why not consider an unset image a validation error instead? The latter feels like a better failure mode to me.
If there's some explanation that I've missed, I suggest that we explain the decision in a comment.

Reply:
01f4ed0 and daa1374

The idea was sorta that you can successfully load no image by doing nothing. This allowed for some laziness downstream, which originally called this with no conditions:

```
	extraImagePlugin := loadimage.NewBuilder().WithImage(os.Getenv(imageOverrideEnvVar)).Build()
	builder := environments.NewBuilder().WithAddons(extraImagePlugin)
```
That envvar may be unset, and the noop version allowed you to not check anything and run with it anyway. In retrospect, more confusing to read that downstream anyway, so now it actually errors and I'll make downstream check what it should do more strictly.